### PR TITLE
fix: [PIE-4750]: Change of deployment type does not update infrastructure type thumbnails

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Change of deployment type does not update infrastructure type thumbnails

#### Screenshots

Now:

https://user-images.githubusercontent.com/96711906/184286805-dfdd3023-4571-443a-9d16-5372938f4d3b.mov

https://user-images.githubusercontent.com/96711906/184286851-bc88b25e-62ec-4474-b3f8-1ab57397b65c.mov

https://user-images.githubusercontent.com/96711906/184286919-66131da2-0a4f-4287-87cd-c4514b97acb3.mov

Before:

https://user-images.githubusercontent.com/96711906/183825824-ebde8ef3-3476-48e2-81e2-2bb53849187a.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10493)

**Review by vkbansal-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraDefinition/DeployInfraDefinition.tsx [hunk 1]
```diff
@@ -599,9 +642,15 @@ export default function DeployInfraDefinition(props: React.PropsWithChildren<unk
...
       )}
-      {!isAzureWebAppDeploymentType(selectedDeploymentType) && (
+      {!(
+        isAzureWebAppDeploymentType(selectedDeploymentType) ||
+        isAzureWebAppInfrastructureType(selectedInfrastructureType)
+      ) && (
         <Card className={stageCss.sectionCard}>
-          {!isServerlessDeploymentType(selectedDeploymentType) && (
+          {!(
+            isServerlessDeploymentType(selectedDeploymentType) ||
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraDefinition/DeployInfraDefinition.tsx:651]: If deployment type is not Serverless then how can infra could be serverless ? If I not wrong, I think there is one-to-one mapping.
Maybe, we can just keep one condition to check deployment type.

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraDefinition/DeployInfraDefinition.tsx:651]: In the new redesign, infra is an individual entity and is not associated with a deploymentType. So, to identify a infrastructure belonging to serverless, we need the infrastructure type.

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraDefinition/DeployInfraDefinition.tsx:651]: You can check the 1st and 3rd video to check the difference

**Review by udayvunnam — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**